sdk-rs/src/websocket_program_account_subscriber.rs

```
@@ -0,0 +1,213 @@
+            ..RpcAccountInfoConfig::default()
+        };
+        let config = RpcProgramAccountsConfig {
+            filters: Some(self.options.filters.clone()),
+            account_config,
+            ..RpcProgramAccountsConfig::default()
+        };
+
+        let url = self.url.clone();
+        let mut latest_slot = 0;
+        let on_update = self.on_update.clone();
+        let local = LocalSet::new();
+
+        let pubsub = PubsubClient::new(&url).await?;
+
+        let event_emitter = self.event_emitter.clone();
+
+        local.run_until(async move {
+            
+            let (mut accounts, unsubscribe) = pubsub.program_subscribe(
+                &drift_program::ID,
+                Some(config)
+            ).await.unwrap(); 
+            
+            self.unsubscriber = Some(unsubscribe);
+            
+            while let Some(message) = accounts.next().await {
+                let slot = message.context.slot;
+                if slot >= latest_slot {
+                    latest_slot = slot;
+                    let pubkey = message.value.pubkey;
+                    let account_data = message.value.account.data;
+                    let on_update_clone = on_update.clone();
+                    match Self::decode(account_data) {
+                        Ok(data) => {
+                            let data_and_slot = DataAndSlot { slot, data };
+                            if let Some(ref on_update_callback) = on_update_clone {
+                                on_update_callback(event_emitter.clone(), pubkey, data_and_slot);                          
+                            }
+                        },
+                        Err(e) => {
+                            error!("Error decoding account data: {e}");
+                        }
+                    }
+                } else {
+                    warn!("Received stale data from slot: {slot}");
+                }
+            }
+
+        }).await;
+
```

maybe reconnect or panic here? otherwise the app could keep running with no updates (silent fail)